src/main/java/frc/robot/subsystems/vision/Vision.java

```diff
@@ -0,0 +1,95 @@
+package frc.robot.subsystems.vision;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.photonvision.EstimatedRobotPose;
+import org.photonvision.PhotonCamera;
+import org.photonvision.PhotonPoseEstimator;
+import org.photonvision.PhotonPoseEstimator.PoseStrategy;
+
+import edu.wpi.first.apriltag.AprilTagFieldLayout.OriginPosition;
+import edu.wpi.first.apriltag.AprilTagFields;
+import edu.wpi.first.math.geometry.Transform3d;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import frc.robot.Constants.VisionConstants;
+import frc.robot.Constants.FieldConstants;;
+
+/**
+ * Runnable that gets AprilTag data from PhotonVision.
+ */
+public class Vision implements Runnable {
+
+  private final PhotonPoseEstimator photonPoseEstimator;
+  private final PhotonCamera photonCamera;
+  private final AtomicReference<EstimatedRobotPose> atomicEstimatedRobotPose = new AtomicReference<EstimatedRobotPose>();
+
+  public Vision(PhotonCamera cameraName, Transform3d robotToCamera) {
+    this.photonCamera = cameraName;
+    PhotonPoseEstimator photonPoseEstimator = null;
+
+    try {
+      var layout = AprilTagFields.k2024Crescendo.loadAprilTagLayoutField();
+      layout.setOrigin(OriginPosition.kBlueAllianceWallRightSide); // Sets the Origin to the Blue AllianceWall and will be flipped in the robot thread
+
+      // if (photonCamera != null) {
+      photonPoseEstimator = photonCamera != null
+          ? new PhotonPoseEstimator(layout, PoseStrategy.MULTI_TAG_PNP_ON_COPROCESSOR, photonCamera, robotToCamera)
+          : null;
+      photonPoseEstimator.setMultiTagFallbackStrategy(PoseStrategy.LOWEST_AMBIGUITY);
```

Comment:
This line will throw null pointer exception if photonPoseEstimator remains null on the line above.

Reply:
Yes, sometimes it take photonvision to make code instance of the camera long than usual if network tables is slow.